### PR TITLE
Add minimal audit policy to local-up-cluster

### PR DESCRIPTION
**Pull request by dims**
Change-Id: Ia2cdb5a6a891665ece5365698f2989ba3782a5ec

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add support for generating audit logs.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -565,13 +565,23 @@ function start_apiserver {
...
+EOF
+
     APISERVER_LOG=${LOG_DIR}/kube-apiserver.log
     ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" apiserver ${swagger_arg} ${authorizer_arg} ${priv_arg} ${runtime_config} \
       ${cloud_config_arg} \
       ${advertise_address} \
       ${node_port_range} \
       --v=${LOG_LEVEL} \
       --vmodule="${LOG_SPEC}" \
+      --audit-policy-file=/tmp/kube-audit-policy-file \
```
Comment (liggitt) [inline hack/local-up-cluster.sh]: use LOG_DIR instead of hard-coding /tmp?

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -565,13 +565,23 @@ function start_apiserver {
...
+      --audit-log-path=${LOG_DIR}/kube-apiserver-audit.log \
```
Comment (liggitt) [inline hack/local-up-cluster.sh]: should we set `--audit-log-maxsize` to avoid filling disks with logs?

**Comment by liggitt** [comment]
is the test failure related?

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -565,13 +565,23 @@ function start_apiserver {
       cloud_config_arg="--cloud-provider=external"
     fi
 
+    cat <<EOF > /tmp/kube-audit-policy-file
```
Comment (liggitt) [inline hack/local-up-cluster.sh]: only do this if `$AUDIT_POLICY_FILE` is unset, and set AUDIT_POLICY_FILE to /tmp/kube-audit-policy-file here, and use `--audit-policy-file="${AUDIT_POLICY_FILE}" \` below?

Comment (cblecker) [reply hack/local-up-cluster.sh]: I wish we weren't hardcoding to `/tmp`, but it's all over the file so it's not exactly going to break anyone.

**Review by cblecker — COMMENTED** [review]
No bash concerns. I defer to @liggitt for the specific flags and format of the policy pile.

File: hack/local-up-cluster.sh (hunk 3 above)
Comment (liggitt) [reply hack/local-up-cluster.sh]: I had the same thought, and the same follow up response. I wouldn't do something different just for this file

**Comment by dims** [comment]
@liggitt @cblecker so can we let this in?

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/72487#discussion_r253973766 still applies

**Comment by dims** [comment]
@liggitt whoops! missed that. fixing now.

**Comment by anarchistHH1983** [comment]
Retest

**Pull request merged**